paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -123,14 +124,45 @@ def describe_instances(instance_ids, region=None, instance_filters=None):
 
 
 @register_autoscaling_component('aws_spot_fleet_request', CLUSTER_METRICS_PROVIDER_KEY)
-def spotfleet_metrics_provider(spotfleet_request_id, resource, pool_settings):
-    mesos_state = get_mesos_master().state
+def spotfleet_metrics_provider(spotfleet_request_id, resource, pool_settings, config_folder, dry_run=False):
     sfr = get_sfr(spotfleet_request_id, region=resource['region'])
-    if not sfr or not sfr['SpotFleetRequestState'] == 'active':
-        log.error("Ignoring SFR {0} that does not exist or is not active.".format(spotfleet_request_id))
+    if not sfr or sfr['SpotFleetRequestState'] == 'cancelled':
+        log.error("SFR not found, removing config file.".format(spotfleet_request_id))
+        cleanup_cancelled_sfr_config(spotfleet_request_id, config_folder, dry_run=dry_run)
+        return 0, 0
+    elif sfr['SpotFleetRequestState'] in ['cancelled_running', 'active']:
+        sfr['ActiveInstances'] = get_spot_fleet_instances(spotfleet_request_id, region=resource['region'])
+        resource['sfr'] = sfr
+        error = get_sfr_utilization_error(spotfleet_request_id, resource, pool_settings)
+    else:
+        log.warning("Unexpected SFR state: {0} for {1}, skipping...".format(sfr['SpotFleetRequestState'],
```

Comment:
There is also `modifying` and `submitted`. In those cases doing nothing is a good choice. I'll change this to make it clearer though, and raise on anything not in the list of states amazons docs provides.